Libraries/Microsoft.Teams.Api/Activities/Activity.cs

```diff
@@ -287,18 +287,58 @@ public virtual Activity AddEntity(params IEntity[] entities)
         return this;
     }
 
+    public virtual Activity UpdateEntity(IEntity oldEntity, IEntity newEntity)
+    {
+        if (Entities != null)
+        {
+            Entities.Remove(oldEntity);
+        }
+        else
+        {
+            Entities = [];
+        }
+
+        Entities.Add(newEntity);
+        return this;
+    }
+
+    /// <summary>
+    /// ensures a single root level message entity exists
+    /// </summary>
+    private IMessageEntity EnsureSingleRootLevelMessageEntity()
```

Comment:
Nit: since you are returning the entity maybe `GetRootLevelMessageEntity` would be easier to understand at a glance.